example/server/main.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2023 CloudWeGo Authors
```

Comment:
Add a `go.mod` under `example/` to avoid introducing extra dependency in the main `go.mod`